wherobots/db/cursor.py

```diff
@@ -43,8 +43,32 @@ def __get_results(self) -> Optional[list[Any]]:
         result = self.__queue.get()
         if isinstance(result, DatabaseError):
             raise result
+        type_map = {
+            'object': 'STRING',
+            'int64': 'NUMBER',
+            'float64': 'NUMBER',
+            'datetime64[ns]': 'DATETIME',
+            'timedelta[ns]': 'DATETIME',
+            'bool': 'NUMBER',  # Assuming boolean is stored as number
+            'bytes': 'BINARY',
+        }
         self.__rowcount = len(result)
         self.__results = result
+        if not result.empty:
+            self.__description = [
+                (
+                    col_name, # name
+                    type_map.get(str(result[col_name].dtype), 'STRING'),  # type_code
```

Comment:
What `dtype` do we get for geometry columns? Is it just `bytes` (and therefore, `BINARY`) ?

Reply:
Geometry columns are in WKT format and `str(result[col_name].dtype` returns "object" type. 
This gets mapped to "STRING".